app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/stackTrace/stackTrace.jsx

```diff
@@ -0,0 +1,180 @@
+  },
+});
+
+@connect((state) => ({
+  logItemId: activeLogIdSelector(state),
+  projectId: activeProjectSelector(state),
+  pagination: logPaginationSelector(state),
+  pageProperties: pagePropertiesSelector(state, NAMESPACE),
+  logItems: logItemsSelector(state),
+}))
+@injectIntl
+export class StackTrace extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    projectId: PropTypes.string.isRequired,
+    logItemId: PropTypes.string.isRequired,
+    onChangePage: PropTypes.func.isRequired,
+    pagination: PropTypes.object.isRequired,
+    pageProperties: PropTypes.object.isRequired,
+    logItems: PropTypes.array.isRequired,
+  };
+
+  static checkIfStackTraceItemOnThisPage = (logItems, stackTraceItem) =>
+    !!logItems.find((item) => item.id === stackTraceItem.id);
+
+  static getDerivedStateFromProps(props, state) {
+    return state.activePage !== props.pagination.number
+      ? {
+          activePage: props.pagination.number,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            props.logItems,
+            state.stackTraceItem,
+          ),
+        }
+      : null;
+  }
+
+  state = {
+    stackTraceItem: null,
+    isStackTraceItemOnThisPage: false,
+    activePage: this.props.pagination.number,
+    loading: false,
+  };
+
+  componentDidMount() {
+    this.fetchStackTrace();
+  }
+
+  fetchStackTrace = () => {
+    const { logItems, projectId, logItemId } = this.props;
+    this.setState({
+      loading: true,
+    });
+    fetch(URLS.logItemStackTrace(projectId, logItemId))
+      .then(({ content }) => {
+        const stackTraceItem = (content.length && content[0]) || null;
+        this.setState({
+          stackTraceItem,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            logItems,
+            stackTraceItem,
+          ),
+          loading: false,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          loading: false,
+        });
+      });
+  };
+
+  goToMessageInLog = ({ wasClicked, targetPageNumber }) => {
+    const { isStackTraceItemOnThisPage, stackTraceItem, activePage } = this.state;
+
+    if (isStackTraceItemOnThisPage) {
+      const targetItem = document.getElementById(stackTraceItem.id);
+      targetItem.scrollIntoView({ behavior: 'smooth' });
+      let highlightElement = document.getElementById('highlight');
+      if (!highlightElement) {
+        highlightElement = document.createElement('div');
+        highlightElement.setAttribute('id', 'highlight');
+        highlightElement.classList.add(cx('highlight'));
+        targetItem.appendChild(highlightElement);
+        setTimeout(() => targetItem.classList.add(cx('hide-highlight')), 0);
+      }
+      return;
+    }
+    if (activePage === targetPageNumber) {
+      return;
+    }
+    if (wasClicked) {
+      setTimeout(() => this.goToMessageInLog({ wasClicked: true }), 400);
```

Comment:
What is it for? Very strange `setTimeout` with strange time and flag

Reply:
`StackTrace` component logic was fully rewritten.